### DISCUSSION TITLE
Wiki Font Change Request

### DISCUSSION THREAD

**Discussion by Valkryst** [jagrosh/MusicBot#1256] [Questions] [answered]
I recently had to read through the Wiki to reinstall the bot for one of my servers, and I found the custom font incredibly difficult to read. Could it be changed to a more standard, legible font?

![image](https://user-images.githubusercontent.com/1524476/204895344-ed1cc92c-73c9-48fc-8a45-12967dd4f78f.png)
![image](https://user-images.githubusercontent.com/1524476/204895400-b1cb0b2b-6ad3-4d41-b6d2-a65667b79a9b.png)

**Comment by jagrosh** [comment, marked as answer]
That's not the font the wiki uses. Are you using a browser extension that either blocks the font it tries to use (and that's the default monospace font in your browser), or changes the font that it uses?

**Reply by Valkryst** [reply]
It's definitely possible, I do have a number of blockers installed

I've disabled my _pihole_ and _uBlock Origin_ both together and independently, but the font remains unchanged. I'm also not seeing any blocked network requests in the browser's _Network_ tab.

**Reply by jagrosh** [reply]
None of the fonts in the chain of backup fonts should look anything like that, so either your browser's default monospace font looks like that, or some extension is changing the font. Try inspecting the elements with dev tools to see what font it's using.

**Reply by jagrosh** [reply]
Also what OS and browser are you using? I have never seen anyone else run into this issue.

**Reply by Valkryst** [reply]
Currently running Windows 10 and accessing it with FireFox. Inspecting one of the `<code>` elements shows this list of fonts: `"", SFMono-Regular, Consolas, Menlo, monospace`

I also checked the browser's font settings, but they seem standard:

![image](https://user-images.githubusercontent.com/1524476/204898625-ae6e1548-93d3-4ffe-ad0f-af86709ff0ae.png)

The font could be coming from one of my installed font files, it does look familiar 🤔

**Reply by jagrosh** [reply]
I also checked the font list and none of those (SFMono-Regular, Consolas, Menlo, monospace) should look like that, so it's extremely likely that it's something you have installed.

**Reply by jagrosh** [reply]
If you try opening the page in a different browser, you may be able to tell if you have something installed in Firefox or if it's your default Windows monospace font. I've viewed the page from multiple separate PCs all running Win10 & Firefox and it renders normally on all of them.

**Reply by Valkryst** [reply]
Looks like it swaps to one of the regular fonts, if I get rid of the `--md-code-font-family` CSS variable. I'll have to check around and see why that would be pulling a random font like this

![image](https://user-images.githubusercontent.com/1524476/204899808-0314684f-a589-4aad-a3d4-1245f7f28bf5.png)

**Reply by Valkryst** [reply]
> If you try opening the page in a different browser, you may be able to tell if you have something installed in Firefox or if it's your default Windows monospace font. I've viewed the page from multiple separate PCs all running Win10 & Firefox and it renders normally on all of them.

Definitely a Firefox-specific issue, it renders properly on Edge

**Reply by jagrosh** [reply]
[This page](https://www.w3schools.com/tags/tag_code.asp) has an example of the `code` tag and might also be of assistance to you if you want to test other pages that use code elements.

**Reply by Valkryst** [reply]
Looks like the issue can be reproduced on that page as well. It renders correctly, but changing the `font-family` from `Consolas,Menlo,"courier new",monospace` to `"",Consolas,Menlo,"courier new",monospace` will cause the issue to appear. I think it's some odd behaviour with setting the font to `""`

**Reply by jagrosh** [reply]
If you inspect the code element, you can go to the "Fonts" tab (by default on the far right, next to tabs like "Layout", "Computed", "Changes") and see the name of the font being used, as well as all fonts on the page. Then, check your default fonts (Settings > Language & Appearance > Fonts > Advanced) and see if you can find where that's being set.

I'm not sure why the autogen tool is putting an empty string in the variable, and I'm not really sure what the correct browser behavior is to handle that. Maybe somehow this font is installed on your system with no name, so the empty string is defaulting to that, when a normal setup would just skip over it?

**Reply by Valkryst** [reply]
The font is called `Tele-Marines` and it's listed as a `System` font, so it's definitely installed somewhere. I can't seem to find it within my installed fonts, at least not yet.

Within FF's font settings, none of the languages have `Tele-Marines` set as a default font. Even more bizarrely, disabling `Allow pages to choose their own fonts, instead of your selections above` (an option in FF's font settings) fixes the issue.

So, it seems like the browser thinks that the site is specifying `Tele-Marines` when it clearly isn't

**Reply by Valkryst** [reply]
Managed to find the font, after some searching. Deleting it from my list of fonts resolved the issue, though it's still a bit puzzling why it would select that specific font in this case 🤔 

Thanks for all of your help looking into this!
